### PR TITLE
Remove ban system, added response whether a server is official or not…

### PR DESCRIPTION
**Pull request by murpii**
… and quick join links and a spam filter for t.me links

### CODE REVIEW

File: cogs/moderator.py [hunk 1]
```diff
@@ -214,6 +53,70 @@ async def on_message(self, message: discord.Message):
             await asyncio.sleep(60 * 60)
             await author.remove_roles(muted_role)
 
+    @commands.Cog.listener('on_message')
+    async def link_filter(self, message: discord.Message):
+
+        if message.guild is None or message.guild.id != GUILD_DDNET or message.channel.id not in (CHAN_DEV, CHAN_WIKI):
```
Comment (def-) [inline cogs/moderator.py:59]:
```suggestion
        if not message.guild or message.guild.id != GUILD_DDNET or message.channel.id not in (CHAN_DEV, CHAN_WIKI):
```

File: cogs/moderator.py [hunk 2]
```diff
@@ -214,6 +53,70 @@ async def on_message(self, message: discord.Message):
...
+            return
+
+        link_pattern = re.compile(r'https?:\/\/(www\.)?t\.me', re.IGNORECASE)
+        if link_pattern.search(message.content):
+            await message.delete()
+
+    @commands.Cog.listener('on_message')
+    async def server_link(self, message: discord.Message):
+        if message.guild is None or message.guild.id != GUILD_DDNET or message.channel.id != CHAN_REPORTS \
```
Comment (def-) [inline cogs/moderator.py:68]:
```suggestion
        if not message.guild or message.guild.id != GUILD_DDNET or message.channel.id != CHAN_REPORTS \
```

File: cogs/moderator.py [hunk 3]
```diff
@@ -214,6 +53,70 @@ async def on_message(self, message: discord.Message):
...
+                        all_servers += server_lists
+            return all_servers
+
+        ddnet = extract_servers(jsondata, ['DDNet', 'Test', 'Tutorial'], "ddnet")
+        ddnetpvp = extract_servers(jsondata, ['Block', 'Infection', 'iCTF', 'gCTF', 'Vanilla', 'zCatch',
+                                              'TeeWare', 'TeeSmash', 'Foot', 'xPanic', 'Monster'], "ddnet")
+        nobyfng = extract_servers(jsondata, ['FNG'], "ddnet")
+        kog = extract_servers(jsondata, ['Gores', 'TestGores'], "kog")
+
+        ipaddr = re.compile(r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}:\d{1,4}')
```
Comment (def-) [inline cogs/moderator.py:96]: this only works for ipv4, not ipv6

**Review by def- — APPROVED** [review]

File: cogs/moderator.py (hunk 1 above)
Comment (heinrich5991) [reply cogs/moderator.py:59]: From PEP8:
> Comparisons to singletons like `None` should always be done with `is` or `is not`, never the equality operators.
>
> Also, beware of writing `if x` when you really mean if `x is not None` – e.g. when testing whether a variable or argument that defaults to None was set to some other value. The other value might have a type (such as a container) that could be false in a boolean context!

Comment (murpii) [reply cogs/moderator.py:59]: I can remove that part entirely as it's not needed anyway.

File: cogs/moderator.py (hunk 3 above)
Comment (def-) [reply cogs/moderator.py:96]: Yes, for example `ddnet://[::1]:8303` is local server.

Comment (murpii) [reply cogs/moderator.py:96]: I was about to add support for IPv6 addresses to the code, but I realized that there are no servers with IPv6 addresses available in "https://info2.ddnet.tw/info".

Comment (def-) [reply cogs/moderator.py:96]: Trashmap runs with ipv6 I think.

Comment (murpii) [reply cogs/moderator.py:96]: I don't think we have them in "https://info2.ddnet.tw/info". Guess I could try using master1 to test it instead but I'd prefer to wait until we have our own servers with ipv6 instead.

Comment (def-) [reply cogs/moderator.py:96]: Oh right. We had, but removed them again, people kept wondering why they can't connect.

**Pull request merged**